serve-jupyterlab-datascience/tests/test_api.py

```diff
@@ -0,0 +1,210 @@
+"""Tests of the jupyter-lab image using API calls."""
+
+import os
+import time
+import datetime
+import uuid
+import json
+import requests
+from requests.exceptions import ConnectionError
+from websocket import create_connection
+import pytest
+import docker
+
+
+# Settings
+PORT = 8888  # the jupyter-lab port
+HOST = f"localhost:{PORT}"  # the host
+NOTEBOOK_PATH = "tests/basic.ipynb"  # the relative path to the jupyter test notebook
+TOKEN = "a268cff61a101aaefe53b02b5a787ddfc0e07d4119154bff"  # the token to use
+TIMEOUT_CALL = 5  # the timeout in seconds of the client request call
+
+
+client = docker.from_env()
+
+container = client.containers.run(
+    os.environ["IMAGE_NAME"],
+    command=f"start-notebook.sh --NotebookApp.token='{TOKEN}'",
+    ports={
+        f"{PORT}/tcp": PORT,
+    },
+    detach=True,
+)
+
+time.sleep(20)
+
+container.reload()
```

Comment:
This could have been a session level fixture https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session

Not insisting on changing this, but letting you know, how it could be done